src/providers/local/pool-staking.provider.ts

```diff
@@ -0,0 +1,232 @@
+import { LiquidityGauge } from '@/services/balancer/contracts/contracts/liquidity-gauge';
+import { getAddress } from '@ethersproject/address';
+import { parseUnits } from '@ethersproject/units';
+import { useTokens } from '../tokens.provider';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import useWeb3 from '@/services/web3/useWeb3';
+import { POOLS } from '@/constants/pools';
+
+/**
+ * TYPES
+ */
+type Props = {
+  poolId: string;
+};
+
+/**
+ * PoolStakingProvider
+ *
+ * Fetches data and provides functionality for a specific pool's gauge.
+ */
+const provider = (props: Props) => {
+  /**
+   * STATE
+   */
+  const poolAddress = computed((): string =>
+    getAddressFromPoolId(props.poolId)
+  );
+
+  /**
+   * COMPOSABLES
+   */
+  const { balanceFor } = useTokens();
+  const { account } = useWeb3();
+
+  /**
+   * QUERIES
+   */
+  // Fetches all gauges for this pool (incl. preferential gauge).
+  const poolGaugesQuery = usePoolGaugesQuery(poolAddress);
+  const { data: poolGauges, refetch: refetchPoolGauges } = poolGaugesQuery;
+
+  // Fetches the user's total staked shares for this pool using onchain calls.
+  const poolStakedSharesQuery = usePoolStakedSharesQuery(poolGauges);
+  const {
+    data: _stakedShares,
+    isRefetching: isRefetchingStakedShares,
+    refetch: refetchStakedShares,
+  } = poolStakedSharesQuery;
+
+  // Fetches user's gaugeShare for pool, to be used in boost calculation.
+  const userGaugeSharesQuery = useUserGaugeSharesQuery(poolAddress);
+  const { data: userGaugeShares, refetch: refetchUserGaugeShares } =
+    userGaugeSharesQuery;
+
+  // Fetches user's boost value for this pool.
+  const userBoostsQuery = useUserBoostsQuery(userGaugeShares);
+  const { data: boostsMap, refetch: refetchUserBoosts } = userBoostsQuery;
+
+  /**
+   * COMPUTED
+   */
+  const isLoading = computed(
+    (): boolean =>
+      isQueryLoading(poolGaugesQuery) ||
+      isQueryLoading(poolStakedSharesQuery) ||
+      isQueryLoading(userGaugeSharesQuery) ||
+      isQueryLoading(userBoostsQuery)
+  );
+
+  const preferentialGaugeAddress = computed(
+    (): string | undefined | null =>
+      poolGauges.value?.pool?.preferentialGauge?.id
+  );
+
+  // Is it possible to stake this pool's BPT?
+  const isStakablePool = computed(
+    (): boolean =>
+      (poolGauges.value?.liquidityGauges || [])[0]?.id !== undefined &&
+      POOLS.Stakable.AllowList.includes(props.poolId)
+  );
+
+  // User's staked shares for pool (onchain data).
+  const stakedShares = computed((): string => _stakedShares.value || '0');
+
+  // User's boost value for this pool
+  const boost = computed((): string => {
+    if (!boostsMap.value) return '1';
+
+    return boostsMap[props.poolId];
+  });
+
+  // Does the user have a balance in a non-preferential gauge
+  const hasNonPrefGaugeBalance = computed((): boolean => {
+    if (!userGaugeShares.value) return false;
+    if (!preferentialGaugeAddress.value) return false;
```

Comment:
if (!userGaugeShares.value || !preferentialGaugeAddress.value)

Reply:
I did this on purpose because I found it a bit more readable to read these guards independently. But I agree a one-liner works just as well here.